consultation_analyser/consultations/views/answers.py

```diff
@@ -50,3 +52,93 @@ def index(
     }
 
     return render(request, "consultations/answers/index.html", context)
+
+
+@login_required  # TODO: check if we still need this
+@user_can_see_consultation
+def show(
+    request: HttpRequest,
+    consultation_slug: str,
+    question_slug: str,
+    response_id: int,
+):
+    consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
+    question = get_object_or_404(models.Question, slug=question_slug, consultation=consultation)
+    response = get_object_or_404(models.Answer, id=response_id)
+
+    # TODO: update with QuestionPart
+    all_theme_mappings = models.ThemeMapping.get_latest_theme_mappings_for_question_part(
+        part=response.question_part
+    )
+    all_themes = models.Theme.objects.filter(id__in=all_theme_mappings.values("theme"))
+    existing_themes = models.ThemeMapping.objects.filter(answer=response).values_list(
+        "theme", flat=True
+    )
+
+    if request.method == "POST":
+        requested_themes = request.POST.getlist("theme")
+        existing_mappings = models.ThemeMapping.objects.filter(answer=response)
+
+        # themes to delete
+        existing_mappings.exclude(theme_id__in=requested_themes).delete()
+
+        # themes to add
+        themes_to_add = set(requested_themes).difference([str(theme) for theme in existing_themes])
+        for theme_id in themes_to_add:
+            models.ThemeMapping.objects.create(
+                answer=response, theme_id=theme_id, reason="User added theme"
+            )
+
+        # flag
+        response.is_theme_mapping_audited = True
+        response.save()
+
+        return redirect(
+            "show_next_response", consultation_slug=consultation_slug, question_slug=question_slug
+        )
+
+    elif request.method == "GET":
+        context = {
+            "consultation_name": consultation.title,
+            "consultation_slug": consultation_slug,
+            "question": question,
+            "response": response,
+            "all_themes": all_themes,
+            "existing_themes": existing_themes,  #  update this key
+            "date_created": datetime.strftime(response.created_at, "%d %B %Y"),
+        }
+
+        return render(request, "consultations/answers/show.html", context)
+
+
+@user_can_see_consultation
+@login_required
+def show_next(request: HttpRequest, consultation_slug: str, question_slug: str):
+    consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
+    question = get_object_or_404(models.Question, slug=question_slug, consultation=consultation)
+
```

Comment:
There is at most one free text `QuestionPart` per `Question` i.e. at most one `QuestionPart` with themes. Can we use this to make logic simpler?

Something along the lines of:
```
question_part_with_themes = question.questionpart_set.get(type=freetext)
```
(With proper notation.)

Not sure where we should deal with the fact that there might be no free-text part? Maybe `show_next` should only be linked to for questions where there is a free-text part?

Reply:
I've made a view for the case where there are no answers to review (either questions without a free-text part or where all of them have been reviewed)

Reply:
The questions list page that I created only lists free text questions, so that limits the questions that we will "show_next" from.